tests/integration/test_cli.py

```diff
@@ -270,11 +270,17 @@ def test_table_output(self):
     def test_version(self):
         p = aws('--version')
         self.assertEqual(p.rc, 0)
-        # The version is wrote to standard out for Python 3.4 and
-        # standard error for other Python versions.
-        version_output = p.stderr.startswith('aws-cli') or \
-            p.stdout.startswith('aws-cli')
-        self.assertTrue(version_output, p.stderr)
+        version_output = p.stdout
+
+        # There should be four components: aws-cli version, Python version,
+        # platform version, and installation source. The CLI and Python
+        # version have fixed format; the platform and install source can
+        # change based on the system they are invoked.
+        self.assertTrue(len(version_output.split(" ")) == 4, p.stderr)
+
+        self.assertTrue(version_output.startswith('aws-cli'), p.stderr)
+        self.assertTrue("Python/" in version_output, p.stderr)
```

Comment:
I wonder if we should make these assertions a regex? I think it will make the assertion a bit more exact and self documenting.